imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -308,8 +310,9 @@ def get_array_dims(self) -> None:
             "dfm_1d": self.dfm.get_number_1d_nodes(),
             "dfm_2d": self.dfm.get_number_2d_nodes(),
         }
+        self.coupling.validate_sprinkling_settings()
 
-        if self.coupling.enable_sprinkling:
+        if self.coupling.enable_sprinkling():
```

Comment:
the assert is for mypy. the type of self.coupling.mf6_msw_well_pkg is Optional[str]  but function self.mf6.get_sprinkling expects a str. With the assert we tell mypy that self.coupling.mf6_msw_well_pkg is actually a str. It's unfortunate, but otherwise we get a mypy error there